xclim/sdba/adjustment.py

```diff
@@ -1204,6 +1212,279 @@ def _adjust(
+        base_kws = base_kws if base_kws is not None else {}
+        adj_kws = adj_kws if adj_kws is not None else {}
+        base_kws.setdefault("nquantiles", 20)
+        base_kws.setdefault("group", Grouper("time", 1))
+        adj_kws.setdefault("interp", "nearest")
+        adj_kws.setdefault("extrapolation", "constant")
+
+        if np.isscalar(base_kws["nquantiles"]):
+            base_kws["nquantiles"] = equally_spaced_nodes(base_kws["nquantiles"])
+        if isinstance(base_kws["group"], str):
+            base_kws["group"] = Grouper(base_kws["group"], 1)
+        if base_kws["group"].name == "time.month":
+            NotImplementedError(
+                "Received `group==time.month` in `base_kws`. Monthly grouping is not currently supported in the MBCn class."
+            )
+        # stack variables and prepare rotations
+        if rot_matrices is None:
+            pts_dim = xr.core.utils.get_temp_dimname(
+                set(ref.dims).union(hist.dims), "multivar"
+            )
+        else:
+            pts_dim = rot_matrices.attrs["crd_dim"]
+            rot_dim = rot_matrices.attrs["new_dim"]
+        ref = stack_variables(ref, pts_dim)
+        hist = stack_variables(hist, pts_dim)
+        if rot_matrices is None:
+            rot_dim = xr.core.utils.get_temp_dimname(
+                set(ref.dims).union(hist.dims), pts_dim + "_prime"
+            )
+            rot_matrices = rand_rot_matrix(
+                ref[pts_dim], num=n_iter, new_dim=rot_dim
+            ).rename(matrices="iterations")
+        pts_dims = (pts_dim, rot_dim)
+
+        # time indices corresponding to group and windowed group
+        # used to divide datasets as map_blocks or groupby would do
+        g_idxs, gw_idxs = time_block_indices(ref.time, base_kws["group"])
+
+        # training, obtain adjustment factors of the npdf transform
+        ds = xr.Dataset(dict(ref=ref, hist=hist))
+        params = {
+            "quantiles": base_kws["nquantiles"],
+            "interp": adj_kws["interp"],
+            "extrapolation": adj_kws["extrapolation"],
+            "pts_dims": pts_dims,
+            "n_escore": n_escore,
+        }
+        out = mbcn_train(
+            ds, rot_matrices=rot_matrices, g_idxs=g_idxs, gw_idxs=gw_idxs, **params
+        )
+        params["group"] = base_kws["group"]
+
+        # postprocess
+        out["rot_matrices"] = rot_matrices
+        out["g_idxs"] = g_idxs
+        out["gw_idxs"] = gw_idxs
+
+        out.af_q.attrs.update(
+            standard_name="Adjustment factors",
+            long_name="Quantile mapping adjustment factors",
+        )
+        obj = cls(
+            _trained=True,
+            hist_calendar=get_calendar(hist),
+            train_units=train_units,
+            **params,
+        )
+        obj.set_dataset(out)
+        return obj
+
+    def adjust(
+        self,
+        sim: xr.Dataset,
+        *,
+        ref: xr.Dataset | None = None,
+        hist: xr.Dataset | None = None,
```

Comment:
The structure here is not ideal: 

1. It needs `sim, ref,hist` but `ref,hist` were already needed in `train`. This is because I only do the npdf training in the training portion, so still need `ref,hist` after that.
2. Since it's still a BaseAdjust class, I think it expects only `sim` as an arg, but `ref,hist` are really not optional here, just a constraint of the class structrure. Perhaps I should define another class type.